Distributed Lock Redis tests do not run on Mac

<!-- If you need to report a security issue please visit https://docs.dapr.io/operations/support/support-security-issues -->

Distributed lock API tests for Redis do not work on Mac (Intel and Arm). Please fix so that we can repair the local development experience for contributors.

cc @seeflood
https://github.com/dapr/components-contrib/pull/1574

@seeflood it does not work for @ItalyPaleAle on Intel, and for me on ARM (M1)

```
--- FAIL: TestStandaloneRedisLock_InitError (0.09s)
    --- FAIL: TestStandaloneRedisLock_InitError/error_when_no_host (0.02s)
        standalone_test.go:60:
            	Error Trace:	standalone_test.go:60
            	Error:      	An error is expected but got nil.
            	Test:       	TestStandaloneRedisLock_InitError/error_when_no_host
FAIL
FAIL	github.com/dapr/components-contrib/lock/redis	0.108s
```

```
CGO_ENABLED=0 go test -v --count=1 ./lock/...
?   	github.com/dapr/components-contrib/lock	[no test files]
=== RUN   TestStandaloneRedisLock_InitError
=== RUN   TestStandaloneRedisLock_InitError/error_when_connection_fail
=== RUN   TestStandaloneRedisLock_InitError/error_when_no_host
    standalone_test.go:60:
        	Error Trace:	standalone_test.go:60
        	Error:      	An error is expected but got nil.
        	Test:       	TestStandaloneRedisLock_InitError/error_when_no_host
=== RUN   TestStandaloneRedisLock_InitError/error_when_wrong_MaxRetries
--- FAIL: TestStandaloneRedisLock_InitError (0.08s)
    --- PASS: TestStandaloneRedisLock_InitError/error_when_connection_fail (0.07s)
    --- FAIL: TestStandaloneRedisLock_InitError/error_when_no_host (0.01s)
    --- PASS: TestStandaloneRedisLock_InitError/error_when_wrong_MaxRetries (0.00s)
=== RUN   TestStandaloneRedisLock_TryLock
--- PASS: TestStandaloneRedisLock_TryLock (0.00s)
FAIL
FAIL	github.com/dapr/components-contrib/lock/redis	0.089s
FAIL
```

```
 bverst@Bernds-MBP  ~/src/dapr/components-contrib   release-1.8  CGO_ENABLED=0 go test -v --count=1 ./lock/...
?   	github.com/dapr/components-contrib/lock	[no test files]
=== RUN   TestStandaloneRedisLock_InitError
=== RUN   TestStandaloneRedisLock_InitError/error_when_connection_fail
=== RUN   TestStandaloneRedisLock_InitError/error_when_no_host
=== RUN   TestStandaloneRedisLock_InitError/error_when_wrong_MaxRetries
--- PASS: TestStandaloneRedisLock_InitError (0.15s)
    --- PASS: TestStandaloneRedisLock_InitError/error_when_connection_fail (0.07s)
    --- PASS: TestStandaloneRedisLock_InitError/error_when_no_host (0.08s)
    --- PASS: TestStandaloneRedisLock_InitError/error_when_wrong_MaxRetries (0.00s)
=== RUN   TestStandaloneRedisLock_TryLock
--- PASS: TestStandaloneRedisLock_TryLock (0.00s)
PASS
ok  	github.com/dapr/components-contrib/lock/redis	0.158s
 bverst@Bernds-MBP  ~/src/dapr/components-contrib   release-1.8  dapr init
⌛  Making the jump to hyperspace...
ℹ️  Container images will be pulled from Docker Hub
ℹ️  Installing runtime version 1.7.4
↗  Downloading binaries and setting up components...
Dapr runtime installed to /Users/bverst/.dapr/bin, you may run the following to add it to your path if you want to run daprd directly:
    export PATH=$PATH:/Users/bverst/.dapr/bin
✅  Downloading binaries and setting up components...
✅  Downloaded binaries and completed components set up.
ℹ️  daprd binary has been installed to /Users/bverst/.dapr/bin.
ℹ️  dapr_placement container is running.
ℹ️  dapr_redis container is running.
ℹ️  dapr_zipkin container is running.
ℹ️  Use `docker ps` to check running containers.
✅  Success! Dapr is up and running. To get started, go here: https://aka.ms/dapr-getting-started
 bverst@Bernds-MBP  ~/src/dapr/components-contrib   release-1.8  CGO_ENABLED=0 go test -v --count=1 ./lock/...
?   	github.com/dapr/components-contrib/lock	[no test files]
=== RUN   TestStandaloneRedisLock_InitError
=== RUN   TestStandaloneRedisLock_InitError/error_when_connection_fail
=== RUN   TestStandaloneRedisLock_InitError/error_when_no_host
    standalone_test.go:60:
        	Error Trace:	standalone_test.go:60
        	Error:      	An error is expected but got nil.
        	Test:       	TestStandaloneRedisLock_InitError/error_when_no_host
=== RUN   TestStandaloneRedisLock_InitError/error_when_wrong_MaxRetries
--- FAIL: TestStandaloneRedisLock_InitError (0.07s)
    --- PASS: TestStandaloneRedisLock_InitError/error_when_connection_fail (0.07s)
    --- FAIL: TestStandaloneRedisLock_InitError/error_when_no_host (0.01s)
    --- PASS: TestStandaloneRedisLock_InitError/error_when_wrong_MaxRetries (0.00s)
=== RUN   TestStandaloneRedisLock_TryLock
--- PASS: TestStandaloneRedisLock_TryLock (0.00s)
FAIL
FAIL	github.com/dapr/components-contrib/lock/redis	0.086s
FAIL
```

--- a/lock/redis/standalone.go
+++ b/lock/redis/standalone.go
@@ -63,6 +63,10 @@ func (r *StandaloneRedisLock) InitLockStore(metadata lock.Metadata) error {
 		return err
 	}
 	r.metadata = m
+	// must have `redisHost`
+	if metadata.Properties["redisHost"] == "" {
+		return fmt.Errorf("[standaloneRedisLock]: InitLockStore error. redisHost is empty")
+	}
 	// no failover
 	if needFailover(metadata.Properties) {
 		return fmt.Errorf("[standaloneRedisLock]: InitLockStore error. Failover is not supported")

